libs/orderbook/src/lib/orderbook-manager.tsx

```diff
@@ -0,0 +1,81 @@
+import { AsyncRenderer } from '@vegaprotocol/ui-toolkit';
+import { Orderbook } from './orderbook';
+import { useDataProvider } from '@vegaprotocol/react-helpers';
+import { marketDepthDataProvider } from './market-depth-data-provider';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import type { MarketDepthSubscription_marketDepthUpdate } from './__generated__/MarketDepthSubscription';
+import { compact, updateCompactedData } from './orderbook-data';
+import type { OrderbookData } from './orderbook-data';
+
+interface OrderbookManagerProps {
+  marketId: string;
+  resolution: number;
+}
+
+export const OrderbookManager = ({
+  marketId,
+  resolution,
+}: OrderbookManagerProps) => {
+  const variables = useMemo(() => ({ marketId }), [marketId]);
+  const resolutionRef = useRef(resolution);
+  const lastUpdateRef = useRef(new Date().getTime());
+  const [orderbookData, setOrderbookData] = useState<OrderbookData[] | null>(
```

Comment:
In data provider only raw data is stored. It's compacted in the component and stored as orderbookData.ref.
This allows to:
 - use same data in more than one component (ie grapth which uses market depth also)
 - change resolution of price level grouping without need of refetching initial data
 
 It could be done in data provider getData and update methods, but I find this way much more  effective